File: src/cli/find.go
package cli

import (
	"fmt"
)

// Find wraps find commands
type Find struct {
	directory string
	maxdepth  *int
	fileType  string
	name      string
}

// NewFind creates a new Find command builder
func NewFind() *Find {
	return &Find{}
}

// Directory sets directory to search
func (f *Find) Directory(path string) *Find {
	f.directory = path
	return f
}

// Maxdepth sets maxdepth
func (f *Find) Maxdepth(depth int) *Find {
	f.maxdepth = &depth
	return f
}

// Type sets file type
func (f *Find) Type(fileType string) *Find {
	f.fileType = fileType
	return f
}

// Name sets name pattern
func (f *Find) Name(pattern string) *Find {
	f.name = pattern
	return f
}

// Delete generates find command with -delete action
func (f *Find) Delete() string {
	if f.directory == "" {
		panic("Directory must be set")
	}
	cmd := fmt.Sprintf("find %s", quote(f.directory))
	if f.maxdepth != nil {
		cmd += fmt.Sprintf(" -maxdepth %d", *f.maxdepth)
	}
	if f.fileType != "" {
		cmd += fmt.Sprintf(" -type %s", f.fileType)
	}
	if f.name != "" {
		patternEscaped := escapeSingleQuotes(f.name)
		cmd += fmt.Sprintf(" -name '%s'", patternEscaped)
	}
	cmd += " -delete || true"
	return cmd
}

// Count generates find command that counts matching files
func (f *Find) Count() string {
	if f.directory == "" {
		panic("Directory must be set")
	}
	cmd := fmt.Sprintf("find %s", quote(f.directory))
	if f.maxdepth != nil {
		cmd += fmt.Sprintf(" -maxdepth %d", *f.maxdepth)
	}
	if f.fileType != "" {
		cmd += fmt.Sprintf(" -type %s", f.fileType)
	}
	if f.name != "" {
		patternEscaped := escapeSingleQuotes(f.name)
		cmd += fmt.Sprintf(" -name '%s'", patternEscaped)
	}
	cmd += " -print | wc -l"
	return cmd
}

File: src/CLI/Find.cs
using System;

namespace Enva.CLI;

public class Find
{
    private string? directory;
    private int? maxdepth;
    private string? fileType;
    private string? name;

    public static Find NewFind()
    {
        return new Find();
    }

    public Find Directory(string path)
    {
        directory = path;
        return this;
    }

    public Find Maxdepth(int depth)
    {
        maxdepth = depth;
        return this;
    }

    public Find Type(string fileTypeValue)
    {
        fileType = fileTypeValue;
        return this;
    }

    public Find Name(string pattern)
    {
        name = pattern;
        return this;
    }

    public string Delete()
    {
        if (string.IsNullOrEmpty(directory))
        {
            throw new InvalidOperationException("Directory must be set");
        }
        string cmd = $"find {Quote(directory)}";
        if (maxdepth.HasValue)
        {
            cmd += $" -maxdepth {maxdepth.Value}";
        }
        if (!string.IsNullOrEmpty(fileType))
        {
            cmd += $" -type {fileType}";
        }
        if (!string.IsNullOrEmpty(name))
        {
            string patternEscaped = EscapeSingleQuotes(name);
            cmd += $" -name '{patternEscaped}'";
        }
        cmd += " -delete || true";
        return cmd;
    }

    public string Count()
    {
        if (string.IsNullOrEmpty(directory))
        {
            throw new InvalidOperationException("Directory must be set");
        }
        string cmd = $"find {Quote(directory)}";
        if (maxdepth.HasValue)
        {
            cmd += $" -maxdepth {maxdepth.Value}";
        }
        if (!string.IsNullOrEmpty(fileType))
        {
            cmd += $" -type {fileType}";
        }
        if (!string.IsNullOrEmpty(name))
        {
            string patternEscaped = EscapeSingleQuotes(name);
            cmd += $" -name '{patternEscaped}'";
        }
        cmd += " -print | wc -l";
        return cmd;
    }

    private string Quote(string s)
    {
        if (s.Contains(" ") || s.Contains("$") || s.Contains("'"))
        {
            return $"'{s.Replace("'", "'\"'\"'")}'";
        }
        return s;
    }

    private string EscapeSingleQuotes(string value)
    {
        return value.Replace("'", "'\"'\"'");
    }
}
